### PR TITLE
Install powershell and azcli on fedora 39 and 40

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

File: src/fedora/40/amd64/Dockerfile [hunk 1]
```diff
@@ -19,7 +19,7 @@ RUN dnf --setopt=install_weak_deps=False install -y \
 
 # Add MS package repo.
 COPY  microsoft.asc /tmp
-RUN dnf config-manager --add-repo https://packages.microsoft.com/fedora/39/prod/ && \
+RUN dnf config-manager --add-repo https://packages.microsoft.com/fedora/40/prod/ && \
```
Comment (mmitche) [inline src/fedora/40/amd64/Dockerfile:22]: @MichaelSimons I noticed the F40 dockerfile was referencing the 39 repo. My guess is that wasn't intentional, but if it was I can change it back.

**Review by vseanreesermsft — APPROVED** [review]
lgtm

File: src/fedora/39/amd64/Dockerfile [hunk 2]
```diff
@@ -29,8 +29,18 @@ RUN dnf --setopt=install_weak_deps=False install -y \
         tar \
         procps \
         zip \
+        azure-cli \
```
Comment (mthalman) [inline src/fedora/39/amd64/Dockerfile:32]: Please sort this for both Dockerfiles.

File: src/fedora/39/amd64/Dockerfile [hunk 3]
```diff
@@ -29,8 +29,18 @@ RUN dnf --setopt=install_weak_deps=False install -y \
...
     && dnf clean all
 
+# Install the latest non-preview powershell release.
+RUN LATEST_TAG=$(curl -L https://api.github.com/repos/powershell/powershell/releases/latest | jq -r '.tag_name') \
```
Comment (mthalman) [inline src/fedora/39/amd64/Dockerfile:36]: Looks like you'll need to install `jq` as well.

File: src/fedora/40/amd64/Dockerfile (hunk 1 above)
Comment (mthalman) [reply src/fedora/40/amd64/Dockerfile:22]: This change looks correct.

File: src/fedora/39/amd64/Dockerfile (hunk 3 above)
Comment (mmitche) [reply src/fedora/39/amd64/Dockerfile:36]: It is installed on the final image, maybe it was just there via a dependency. I'll ensure it's explicit.

**Review by mthalman — APPROVED** [review]

**Pull request merged**